app/scripts/modules/core/delivery/executionGroup/execution/ExecutionMarker.tsx

```diff
@@ -45,28 +50,40 @@ export class ExecutionMarker extends React.Component<IExecutionMarkerProps, IExe
   private handleStageClick = () => {
     ReactGA.event({category: 'Pipeline', action: 'Stage clicked (bar)'});
     this.props.onClick(this.props.stage.index);
-  }
+  };
 
   public render() {
-    const stage = this.props.stage;
+    const {stage, application, execution} = this.props;
     const markerClassName = [
       'clickable',
       'stage',
       'execution-marker',
       `stage-type-${stage.type.toLowerCase()}`,
       `execution-marker-${stage.status.toLowerCase()}`,
       this.props.active ? 'active' : '',
+      this.props.previousStageActive ? 'after-active' : '',
       stage.isRunning ? 'glowing' : ''
       ].join(' ');
 
     const TooltipTemplate = stage.labelTemplate;
+    const MarkerIcon = stage.markerIcon;
+    const stageContents = (
+      <div className={markerClassName}
+           style={{width: this.props.width, backgroundColor: stage.color}}
+           onClick={this.handleStageClick}>
+        <MarkerIcon stage={stage}/>
+        <span className="duration">{duration(stage.runningTimeInMs)}</span>
```

Comment:
Just a "something to remember". I do this too, but the "React way" would be to do this `duration()` call somewhere else and set state with the duration (if you want it to rerender every time `stage.runningTimeInMs` changes.